pytorch_pfn_extras/training/extensions/lr_scheduler.py

```diff
@@ -71,12 +71,12 @@ def __init__(
     def __call__(self, manager: ExtensionsManagerProtocol) -> None:
         # https://github.com/pytorch/pytorch/blob/v2.0.1/torch/optim/lr_scheduler.py#L137-L138
         if (
-            self.scheduler.optimizer._step_count < 1
-            and self.wait_for_first_optimizer_step
+            self.wait_for_first_optimizer_step
+            and hasattr(self.scheduler.optimizer.step, "_with_counter")
```

Comment:
A very nit: I think assuming the existence of private APIs is fragile, because this feature will silently stop working if PyTorch renames this attribute. I guess this check is not essentially needed and can be removed?

Reply:
When using the base class of PyTorch's LRScheduler, the addition of the `_step_count` attribute and the addition of a hook for counting are performed at the time of initialization. However, as `ReduceLROnPlateau` does not inherit from this base class, access to these attributes results in an error. Therefore, it is appropriate to perform an existence check for the `_with_counter` attribute, following the code that checks for PyTorch's `lr_scheduler` `_step_count`.

Reply:
Ah got it, thanks! I'd prefer this to be commented somewhere, but it can be done separately 😄

Reply:
I think this may be a very detailed matter and not essential to the functionality of the code, would it be possible to include the link to the comment made on this PR as a comment in the code?